### PR TITLE
Update otel-collector image version and add port 4318 to otel-collector

### PR DESCRIPTION
**Pull request by uralsemih**

### CODE REVIEW

**Comment by adleong** [comment]
Hi @uralsemih.  A few things that you'll need to do to fix up this PR:

* Add a `collector-otlp-http` Server resource to the `tracing-policy.yaml` template
* Run `go test ./... -update` to update the test fixtures
* Run `bin/helm-docs` to update the Helm docs
* Agree to the DCO as described in https://github.com/linkerd/linkerd2/blob/main/CONTRIBUTING.md#developer-certificate-of-origin

**Comment by uralsemih** [comment]
> Hi @uralsemih. A few things that you'll need to do to fix up this PR:
> 
> * Add a `collector-otlp-http` Server resource to the `tracing-policy.yaml` template
> * Run `go test ./... -update` to update the test fixtures
> * Run `bin/helm-docs` to update the Helm docs
> * Agree to the DCO as described in https://github.com/linkerd/linkerd2/blob/main/CONTRIBUTING.md#developer-certificate-of-origin

@adleong 
Do you want me to create another PR for this ?

**Comment by adleong** [comment]
You should be able to push more commits to your branch on your fork to implement the necessary changes.

**Comment by uralsemih** [comment]
I agree to the DCO for all the commits in this PR.

**Comment by uralsemih** [comment]
@adleong Please let me know if there is any missing thing

File: jaeger/charts/linkerd-jaeger/templates/tracing-policy.yaml [hunk 1]
```diff
@@ -19,6 +19,23 @@ spec:
...
+    linkerd.io/extension: jaeger
+    component: collector
+  annotations:
+    {{ include "partials.annotations.created-by" . }}
+spec:
+  podSelector:
+    matchLabels:
+      component: collector
+  port: 4318
+  proxyProtocol: htto
```
Comment (alpeb) [inline jaeger/charts/linkerd-jaeger/templates/tracing-policy.yaml:35]:
```suggestion
  proxyProtocol: HTTP/1
```

**Review by alpeb — CHANGES_REQUESTED** [review]

**Comment by alpeb** [comment]
Fixes #9366

**Comment by alpeb** [comment]
@uralsemih can you please take care of the typo I flagged above?

**Comment by kleimkuhler** [comment]
Checking in @uralsemih do you need anything in order to wrap this up?

**Comment by uralsemih** [comment]
> @uralsemih can you please take care of the typo I flagged above?

Sorry for the late reply. Here I did 10d1cece4291aeb8aca2344656ae6459ae0cd9f2

**Review by adleong — APPROVED** [review]

**Review by alpeb — APPROVED** [review]
LGTM :+1:

**Pull request merged**